packages/python/accern_xyme/util.py

```diff
@@ -573,3 +573,41 @@ def maybe_json_loads(value: str) -> Any:
         return json.loads(escape_str(value))
     except json.JSONDecodeError:
         return None
+
+
+def compute_parallel(
+        tasks: List[int],
+        computing_fn: Callable[[int, threading.RLock], Any],
+        progress_fn: Optional[Callable[..., Any]],
+        max_threads: int) -> List[Any]:
+    a_tasks = list(enumerate(tasks))
+    task_count = 0
+    results = [None] * len(a_tasks)
+    progress_lock = threading.RLock()
+    compute_lock = threading.RLock()
+
+    def runner() -> None:
```

Comment:
error handling is missing -- you cannot let the threads silently die. capture the first exception -- prevent any other computation if an exception is set and reraise the exception